wordpress-mobile/WordPress-Android
# Issue/8829 null smart lock helper

Fixes #8829 

The mSmartLockHelper wasn't being initialized when the user was redirected from a sign-up screen to login. This PR fixes the issue.

To test:
1. Clear app data and open the app
2. Click on Sign up for WordPress.com
3. Click on Sign up with email
4. Enter email of an existing account
5. Click on Next
6. Notice you are redirected to the `LoginMagicLinkRequestFragment` screen
7. Check the AppLog doesn't contain "Internal inconsistency error! mSmartLockHelper found null!" message
---------------
1. Clear app data and open the app
2. Click on Sign In
3. Notice SmartLock account chooser is shown
4. Choose one of the accounts
5. Make sure you are logged in successfully

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
